From e589cc838327803cbbe20baceac29e4a2e281c27 Mon Sep 17 00:00:00 2001
From: "Raphael S. Carvalho" <raphaelsc@cloudius-systems.com>
Date: Mon, 29 Sep 2014 11:07:57 -0300
Subject: [PATCH 2/3] vfs: use va_size instead when calculating st_blocks

st_size field is assigned to va_size, then let's use va_size again
when calculating st_blocks. Better to use the values from the
structure returned by VOP_GETATTR, which is supposed to fill all
the needed fields by vn_stat. Also avoid a needless read into
the underlying vnode.

Signed-off-by: Raphael S. Carvalho <raphaelsc@cloudius-systems.com>
---
 fs/vfs/vfs_vnode.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fs/vfs/vfs_vnode.cc b/fs/vfs/vfs_vnode.cc
index 24c70fb..067112e 100644
--- a/fs/vfs/vfs_vnode.cc
+++ b/fs/vfs/vfs_vnode.cc
@@ -357,7 +357,7 @@ vn_stat(struct vnode *vp, struct stat *st)
 	st->st_mode = mode;
 	st->st_nlink = vap->va_nlink;
 	st->st_blksize = BSIZE;
-	st->st_blocks = vp->v_size / S_BLKSIZE;
+	st->st_blocks = vap->va_size / S_BLKSIZE;
 	st->st_uid = vap->va_uid;
 	st->st_gid = vap->va_gid;
 	st->st_dev = vap->va_fsid;
-- 
1.9.3

